### PR TITLE
Fire UI events when cursor moves or a marker is added

### LINKED ISSUE #3258: Keyboard Navigation: Dispatch event on cursor change
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
Dispatch ui events when the cursor changes. Potential use case: 
1. Update aria live region with information on where the cursor has gone to and where it came from. 
2. Play some kind of earcon when the user has changed cursor locations

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by alschmiedt**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves
#3258

### Proposed Changes
Dispatch a ui event when a cursor changes location or when a marker is placed. 
Also moves getSourceBlock from navigation to astnode because it should have been there all along.

### Reason for Changes
This will be really helpful when we start testing what to say when a user is moving around with the cursor. Also at the summit it seemed pretty common for people to want to do something every time the cursor changes location.

### Test Coverage

Tested on:

### CODE REVIEW

File: core/keyboard_nav/ast_node.js [hunk 1]
```diff
@@ -470,6 +470,23 @@ Blockly.ASTNode.prototype.findFirstFieldOrInput_ = function(block) {
   return null;
 };
 
+/**
+ * Finds the source block of the location of this node.
+ * @return {Blockly.Block} The source block of the location, or null if the node
+ * is of type workspace.
```
Comment (samelhusseini) [inline core/keyboard_nav/ast_node.js]: of type `Blockly.Workspace`.

Comment (samelhusseini) [inline core/keyboard_nav/ast_node.js]: Add visibility

Comment (samelhusseini) [reply core/keyboard_nav/ast_node.js]: (unless you're trying to make this public)

Comment (alschmiedt) [reply core/keyboard_nav/ast_node.js]: Yeah was trying to make it public. I think it is helpful for people trying to create different cursors.

**Pull request merged**